### PR TITLE
Support CLion 2024.2

### PR DESCRIPTION
**Pull request by LeFrosch**
# Checklist

- [ ] I have filed an issue about this change and discussed potential changes with the maintainers.
- [ ] I have received the approval from the maintainers to make this change.
- [ ] This is not a stylistic, refactoring, or cleanup change.

Please note that the maintainers will not be reviewing this change until all checkboxes are ticked. See 
the [Contributions](https://github.com/bazelbuild/intellij#contributions) section in the README for more 
details.

# Discussion thread for this change

Followup to: #6434

# Description of this change
Had to make changes to `CMakeNotificationProviderWrapper` sdkcompat because `CMakeNotificationProvider` was removed in 242.

### CODE REVIEW

File: sdkcompat/v242/com/google/idea/sdkcompat/clion/CMakeNotificationProviderWrapper.java [hunk 1]
```diff
@@ -17,32 +17,20 @@
...
 import com.intellij.openapi.vfs.VirtualFile;
-import com.jetbrains.cidr.cpp.cmake.workspace.CMakeNotificationProvider;
-
-import java.util.function.Function;
 import javax.annotation.Nullable;
-import javax.swing.*;
+import javax.swing.JComponent;
 
-// #api223
+// #api224
```
Comment (tpasternak) [inline sdkcompat/v242/com/google/idea/sdkcompat/clion/CMakeNotificationProviderWrapper.java:27]: @LeFrosch about the notation - this should be the *last IJ version for which the sdkcompat layer should be kept*. In the other words - when we will decide to drop 223, we will grep for #api223 occurrences and delete/inline everything

File: WORKSPACE.bzlmod [hunk 2]
```diff
@@ -291,16 +291,16 @@ http_archive(
     url = CLION_241_URL,
 )
 
-# CLION_242_SHA = "23039b46c172d820a3764e16b2f2f88bf6a2859436ae8ccf6fe5d9c78689b068"
+CLION_242_SHA = "23039b46c172d820a3764e16b2f2f88bf6a2859436ae8ccf6fe5d9c78689b068"
```
Comment (ujohnny) [inline WORKSPACE.bzlmod:294]: let's update to 

```
CLION_242_SHA = "cd018a71bccd28512dc22d8b3824f68ec9450b37c9e87e6c8553d3292acf9d0e"

CLION_242_URL = "https://www.jetbrains.com/intellij-repository/snapshots/com/jetbrains/intellij/clion/clion/242.16677.29-EAP-SNAPSHOT/clion-242.16677.29-EAP-SNAPSHOT.zip"
```

File: cpp/sdkcompat/v242/com/google/idea/blaze/cpp/qsync/OCWorkspaceModifiableModelDisposer.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2023 The Bazel Authors. All rights reserved.
```
Comment (ujohnny) [inline cpp/sdkcompat/v242/com/google/idea/blaze/cpp/qsync/OCWorkspaceModifiableModelDisposer.java:2]: 2024

**Review by ujohnny — APPROVED** [review]
overall ok, let's update to the latest eap and proceed with merge

**Pull request merged**